.github/workflows/docs-gen-and-push.yml

```diff
@@ -25,6 +25,8 @@ jobs:
   debug:
     name: print relevant info
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
```

Comment:
It looks like there is some confusion wrt recent changes, this is undoing #3016. I recommend rebasing onto the latest `main`.